### ISSUE TITLE
Unable connect because error: "The bot's MSA appId or password is incorrect"

### ISSUE THREAD

**Issue by falvessc** [Microsoft/BotFramework-Emulator#253]
Hi everyone!

I have a bot created using Visual Studio 2015 (.NET C#)
This bot is running local, on 3979 port and localhost (look this image below).

![bot running](https://user-images.githubusercontent.com/9569671/28592180-8b84b48c-715e-11e7-8632-f5a4151c775f.png)

About the Web.config file, the AppId and Password are blank.

![webconfig - file](https://user-images.githubusercontent.com/9569671/28592870-c810f30a-7160-11e7-8a6c-973243d06c45.png)

I'm using the lastest stable version of Bot Builder framework (v3.8.5)

![bot builder installed](https://user-images.githubusercontent.com/9569671/28593380-7b7bcd4c-7162-11e7-9e78-dba6a36da4eb.png)

Here show the connection test with problem and is the same port and host (local) running.

![bot framework emulator](https://user-images.githubusercontent.com/9569671/28592117-682ad9da-715e-11e7-9f9f-9c65aec8782a.png)

I was looking to get informations to solve my problem, I'm sorry if there is have another issue open with this problem.
I don't know if is necessary anothers information about the project.
Can you help me to solve this problem?

Thanks a lot.

**Comment by EricDahlvang** [comment]
Hi @falvessc 

Your configuration looks correct.  There was a temporary issue yesterday that may have caused the problem you experienced.  Please try again today, and let us know if you still have issues.

**Comment by falvessc** [comment]
Hi @EricDahlvang 

Unfortunately I tried to run the emulator right now and found the same issue.
I didn't change anything that I report yesterday.

**Comment by EricDahlvang** [comment]
When you first run the emulator, do you see something like the following in the log window?

![image](https://user-images.githubusercontent.com/11055362/28687704-22e91370-72c4-11e7-9a3e-67e5130437af.png)

**Comment by eanders-MS** [comment]
The 401/Unauthorized may be unrelated to MSA. The emulator knows that a 401 can mean failure by the authentication service, and reports it that way, but it's possible for other services, like proxies, to return 401. This may be the case for your network configuration. A few simple things to try:
* Change your bot endpoint in the emulator to https .
* Try 127.0.0.1 instead of localhost.

Less simple, but worth doing:
* Capture a Fiddler trace of your network traffic to determine who is returning 401.

**Comment by falvessc** [comment]
Hi @eanders-MS and @EricDahlvang 

Sorry for the delay.
I fixed the issue yesterday and now is works fine about the bot framework emulator on localhost.
Look this image below.

![bot framework emulator fixed](https://user-images.githubusercontent.com/9569671/28940423-13e01bc4-786b-11e7-9675-8921c7f39f73.png)

I was with NuGet packages problems on my project.
The solution was only uninstall the System.IdentityModel.Tokens.Jwt package with Microsft.Bot.Builder together and install again, because the reference assemblie was with problems.

Actually, right now, I'm using the 4.0.4.403061554 version of System.IdentityMode.Tokens.Jwt and the lastest stable version of Bot Builder framework (v3.9.0).

Thank you for the suggestions and support.
I'll close the issue.

**Comment by eanders-MS** [comment]
Glad to hear you're unblocked, and thank you for letting us now the solution.

**Comment by artemious7** [comment]
Also just had this problem, and updating the Microsft.Bot.Builder nuget package in my bot project has helped.
